tests/phpunit/tests/post.php

```diff
@@ -842,9 +845,16 @@ public function test_wp_insert_post_handle_malformed_post_date( $input, $expecte
 	}
 
 	/**
-	 * @ticket 26798
+	 * Data provider for test_wp_insert_post_handle_malformed_post_date().
+	 *
+	 * @return array[] {
+	 *     Arguments passed to test.
+	 *
+	 *     @type string $input    The input post_date value.
+	 *     @type bool   $expected Whether the post is expected to be inserted.
+	 * }
 	 */
-	public function data_wp_insert_post_handle_malformed_post_date() {
+	public static function data_wp_insert_post_handle_malformed_post_date() {
```

Comment:
I think this is preferable.

```suggestion
	 * @return array<string, array{ date: string, expected: bool }>
	 */
	public static function data_wp_insert_post_handle_malformed_post_date(): array {
```

It would require the arrays in the data provider to be made associative as well.